File: Scala/src/main/scala/PrimaryClasses/K07.scala
package PrimaryClasses
import IExecutable.IExecutable

import scala.collection.mutable

class K07 extends IExecutable {

  def ShowText(): Unit = {
    println("Hello World!")
    println("ようこそ")
    println("Scala言語の世界へ！")
  }

  def GetMaxValue(value1: Int, value2: Int, value3: Int): Int = {
    var temp = if (value1 >= value2) value1 else value2
    temp = if (temp >= value3) temp else value3
    temp
  }

  def GetType(age: Int): Int = {
    age match {
      case x if x <= 0 => 0
      case x if x < 3 || x >= 70 => 1
      case x if x >= 3 && x <= 15 => 2
      case x if x >= 60 && x < 70 => 3
      case _ => 4
    }
  }

  override def Question1(): Unit = {
    var choice = 0
    do {
      print("メッセージを表示しますか？（０：終了する　１：表示する）＞")
      choice = scala.io.StdIn.readInt()
      if (choice == 1) {
        ShowText()
      }
    } while (choice != 0)
    println("終了します。")
  }

  override def Question2(): Unit = {
    var numbers = mutable.MutableList[Int]()
    for (i <- 1 to 3) {
      printf("%dつ目の値を入力してください。＞", i)
      numbers += scala.io.StdIn.readInt()
    }
    val maxValue = GetMaxValue(numbers.head, numbers(1), numbers(2))
    printf("%dつの中で最大値は%dです。\n", numbers.length, maxValue)
  }

  override def Question3(): Unit = {
    var age: Int = 0
    var no: Int = 0
    printf("年齢を入力して下さい。＞")
    age = scala.io.StdIn.readInt()

    no = GetType(age)
    no match {
      case 0 => println("不適切な値が入力されました。")
      case 1 => println("入場料金無料です。")
      case 2 => println("子供料金で半額です。")
      case 3 => println("シニア割引で一割引きです。")
      case _ => println("通常料金です。")
    }
  }

  override def Question4(): Unit = {

  }
}

File: Java/src/com/ecc/gkgo1a/K07.java
package com.ecc.gkgo1a;

import java.util.Scanner;
import java.util.Vector;

public class K07 extends IExecutable {

    private void ShowText() {
        System.out.println("Hello World!");
        System.out.println("ようこそ");
        System.out.println("Java言語の世界へ！");
    }

    private int GetMaxValue(int value1, int value2, int value3) {
        int temp = value1 >= value2 ? value1 : value2;
        temp = temp >= value3 ? temp : value3;
        return temp;
    }

    private int GetType(int age) {
        if (age <= 0)
            return 0;
        else if (age < 3 || age >= 70)
            return 1;
        else if (age >= 3 && age <= 15)
            return 2;
        else if (age >= 60 && age < 70)
            return 3;
        else
            return 4;
    }

    @Override
    public void Question1() {
        int choice;
        try {
            do {
                System.out.print("メッセージを表示しますか？（０：終了する　１：表示する）＞");
                Scanner in = new Scanner(System.in);
                choice = in.nextInt();
                if (choice == 1)
                    ShowText();
            } while (choice != 0);
            System.out.println("終了します。");
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
    }

    @Override
    public void Question2() {
        Vector<Integer> numbers = new Vector<>();
        Scanner in = new Scanner(System.in);
        for (int i = 0; i < 3; i++) {
            System.out.println(String.format("%dつ目の値を入力してください。＞", i + 1));
            numbers.add(in.nextInt());
        }
        System.out.println(String.format("%dつの中で最大値は%dです。", numbers.size(),
                GetMaxValue(numbers.get(0), numbers.get(1), numbers.get(2))));
    }

    @Override
    public void Question3() {
        int age, no;
        System.out.print("年齢を入力して下さい。＞");
        Scanner in = new Scanner(System.in);
        age = in.nextInt();

        no = GetType(age);
        switch (no)
        {
            case 0:
                System.out.println("不適切な値が入力されました。");
                break;
            case 1:
                System.out.println("入場料金無料です。");
                break;
            case 2:
                System.out.println("子供料金で半額です。");
                break;
            case 3:
                System.out.println("シニア割引で１割引きです。");
                break;
            default:
                System.out.println("通常料金です。");
                break;
        }
    }

    @Override
    public void Question4() {

    }
}
